File: src/main/java/dev/lbuddyboy/flash/thread/UserUpdateTask.java
package dev.lbuddyboy.flash.thread;

import dev.lbuddyboy.flash.Flash;
import dev.lbuddyboy.flash.FlashLanguage;
import dev.lbuddyboy.flash.user.User;
import dev.lbuddyboy.flash.util.bukkit.CC;
import org.bukkit.Bukkit;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.List;

public class UserUpdateTask extends Thread {

    @Override
    public void run() {
        for (User user : Flash.getInstance().getUserHandler().getUsers().values()) {
            user.updatePerms();
            user.updateGrants();
        }

        try {
            sleep(10_000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

    }
}

File: src/main/java/dev/lbuddyboy/flash/thread/UserUpdateTask.kt
package dev.lbuddyboy.flash.thread

import dev.lbuddyboy.flash.Flash

class UserUpdateTask : Thread() {
    override fun run() {
        for (user in Flash.instance.userHandler.getUsers().values()) {
            user.updatePerms()
            user.updateGrants()
            if (user.activeRank.isStaff) user.getStaffInfo().playTime = user.getStaffInfo().playTime + 10000
        }
        try {
            sleep(10000)
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
    }
}